[AMBARI-25187] Kerberos operations are shown in service action dropdown when not needed

payert opened the pull request:
Change-Id: I9084f56a1397a080494a5ba34cddd77d2cf59f3a

## What changes were proposed in this pull request?

- Kerberos page template (kerberos.hbs) is fixed to properly reach the "isManualKerberos" and "isKerberosButtonsDisabled" variables in the controller. This way the "REGENERATE KEYTABS" button is removed from the UI.
- Also, the "ACTIONS" service dropdown menu is fixed and the "Regenerate Kerberos" menu item is removed if Kerberos is not managed by Ambari.

## How was this patch tested?
The patch was tested manually. 
- When Kerberos was enabled on the cluster "Manual Kerberos Setup" was chosen.
- As a result the "REGENERATE KEYTABS" button is not shown on the Kerberos
![Screenshot Kerberos page](https://user-images.githubusercontent.com/35402259/55493408-2c6d5c80-5639-11e9-85e4-9a17846645c9.png)
 page.
- In the dropdown service menu the "REGENERATE KEYTABS" menu item is removed.
![Screenshot 2019-04-04 at 16 36 23](https://user-images.githubusercontent.com/35402259/55564407-18d4fb00-56f8-11e9-87e5-596d2f4beb28.png)


- No unit test was added.

Commit:
[AMBARI-25187] Fixed Kerberos page template to disable kerberos related buttons when Ambari is not managing Kerberos identities.

Kerberos page template is fixed to properly reach isManualKerberos and isKerberosButtonsDisabled variables.

Change-Id: I9084f56a1397a080494a5ba34cddd77d2cf59f3a

payert commented:
Team could you please review? @benyoka @zeroflag @g-boros @rlevas @kasakrisz 
Thanks!

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-Github-PullRequest-Builder/5090/
Test PASSed.

rlevas commented:
@payert , If the manual option is chosen, then Ambari is not managing the Kerberos identities. Therefore there is no need for Ambari to regenerate keytab files.  This is why the "Regenerate Keytabs" button is not available.   If it was available, then would would you want it to do?

rlevas commented:
I do not believe that this patch it really related to [AMBARI-25187](https://issues.apache.org/jira/browse/AMBARI-25187).  

@payert , Please indicate what part of that JIRA you think you are solving.

payert commented:
@rlevas , AMBARI-25187 states:
"The "Regenerate Keytabs" option should not be available to the user since Ambari is not managing Kerberos identities. The action doesn't do anything, but the user should not have the option to select it."

The patch removes the "Regenerate Keytabs" button (which was available) when Ambari is not managing Kerberos identities.

Now I am confused what the AMBARI-25187 is about, could you please elaborate ?
Thanks

aBabiichuk commented:
@rlevas as far as I see, the issue here is that Regenerate button is still present even for manual kerberos, because it's handled incorrectly on UI side.

Commit:
Remove "Regenerate Keytabs" menu items from "ACTIONS" dropdown menu when Kerberos is not managed by Ambari

Change-Id: Iff78afbfbe8d49587e4ffa780054887d1e49b2c6

rlevas commented:
@payert , @aBabiichuk .... thanks for the clarification. I misunderstood what this patch was doing.

Commit:
Fix reaching "isManualKerberos" variable

The "isManualKerberos" variable in "host_components.js" is not reachable
via App.get.

Change-Id: I74ad043fbdf5147ec3000778a7caa68051cd1655

payert commented:
I have reviewed the defect description again and realised that the "Actions" dropdown menu should also be fixed. The last updated removed the "regenerate keytabs" menu item if kdc_type is none.

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-Github-PullRequest-Builder/5103/
Test PASSed.

payert commented:
@aBabiichuk , thanks for the review. The `App.get('router.mainAdminKerberosController.isManualKerberos')` does not work here for some reason. Maybe `isManualKerberos` is not yet evaluated at this point.

aBabiichuk commented:
@payert `Em.computed.bool(...)` wouldn't work anyway, this is not the case where you should use computed property

payert commented:
> `Em.computed.bool(...)` wouldn't work anyway, this is not the case where you should use computed property
@aBabiichuk , thanks, what do you suggest then ?

aBabiichuk commented:
@payert to use `App.get('router.mainAdminKerberosController.isManualKerberos')` or some other boolean (if any) which inducates if kerberos is manual or not, making sure that it's set correctly and that `App.HostComponentActionMap.getMap` is called after it's set
